### PR TITLE
1)modified user Schema to have subcollections, 2)added friend request…

### PR DESCRIPTION
**Pull request by gitpugal**
… toggling and 3)adjusted the visibility of darkmode button on mobile screens

### CODE REVIEW

**Comment by narayan954** [comment]
I'll be reviewing tomorrow, you should remove your credentials from the commit.

**Comment by gitpugal** [comment]
Yeah that's fine!

**Comment by gitpugal** [comment]
can we merge now is it ok now?

**Comment by narayan954** [comment]
> can we merge now is it ok now?

checking now

File: src/components/ImgUpload/index.jsx [hunk 1]
```diff
@@ -77,45 +77,54 @@ export default function ImgUpload(props) {
...
-      .catch((err) => {
-        playErrorSound();
-        enqueueSnackbar(err.message, {
-          variant: "error",
+      const postId = postRef.id; // Store post ID in a separate variable
+
+      await db
+        .collection("users")
+        .doc(props.user.uid)
+        .update({
```
Comment (narayan954) [inline src/components/ImgUpload/index.jsx:97]: if the document does not exists, it'll give error, use some property like upsert=true of mongodb

File: src/lib/firebase.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ const firebaseApp = firebase.initializeApp({
   storageBucket: import.meta.env.VITE_DUMMYGRAM_STORAGEBUCKET,
   messagingSenderId: import.meta.env.VITE_DUMMYGRAM_MESSAGINGSENDERID,
   appId: import.meta.env.VITE_DUMMYGRAM_APPID,
-  measurementId: import.meta.env.VITE_DUMMYGRAM_MEASUREMENTID,
+  // measurementId: import.meta.env.VITE_DUMMYGRAM_MEASUREMENTID,
```
Comment (narayan954) [inline src/lib/firebase.js:15]: is this not needed? just curious

File: src/components/ImgUpload/index.jsx (hunk 1 above)
Comment (narayan954) [reply src/components/ImgUpload/index.jsx:97]: set method will fix perhaps, right?

File: src/pages/Profile/index.jsx [hunk 3]
```diff
@@ -49,45 +51,77 @@ function Profile() {
...
+        .delete()
+        .then(() => {
+          db.collection("users")
+            .doc(targetUserUid)
+            .collection("notifications")
+            .doc(currentUserUid)
+            .delete()
+            .then(() => {
+              setFriendRequestSent(false);
+            });
```
Comment (narayan954) [inline src/pages/Profile/index.jsx:71]: add a snackbar telling the status of deleting friendrequest

File: src/pages/Signup/index.jsx [hunk 4]
```diff
@@ -112,16 +113,22 @@ const SignupScreen = () => {
     }
 
     if (submitable) {
-      const usernameDoc = db.doc(`usernames/${username}`);
+      const usernameDoc = db.collection(`users`);
       const batch = db.batch();
```
Comment (narayan954) [inline src/pages/Signup/index.jsx:117]: we are not using batch right? so we can remove it

File: src/pages/Signup/index.jsx [hunk 5]
```diff
@@ -112,16 +113,22 @@ const SignupScreen = () => {
     }
 
     if (submitable) {
-      const usernameDoc = db.doc(`usernames/${username}`);
+      const usernameDoc = db.collection(`users`);
```
Comment (narayan954) [inline src/pages/Signup/index.jsx:116]: we also don't need this variable anymore, we can directly call it in the function below

File: src/pages/Signup/index.jsx [hunk 6]
```diff
@@ -190,31 +197,80 @@ const SignupScreen = () => {
...
-        playSuccessSound();
-        enqueueSnackbar("Signin successful!", {
-          variant: "success",
-        });
-        navigate("/dummygram");
+      .then(async (val) => {
+        setFullName(val?.user?.displayName);
+        setEmail(val?.user?.email);
+        setIsOauthSignUp(true);
+        const usernameDoc = db.collection(`users`);
```
Comment (narayan954) [inline src/pages/Signup/index.jsx:204]: we don't need to store this perhaps, we can call it directly

File: src/components/ImgUpload/index.jsx (hunk 1 above)
Comment (narayan954) [reply src/components/ImgUpload/index.jsx:97]: old users should also be able to work on the app without any error

**Review by narayan954 — CHANGES_REQUESTED** [review]
minor changes requested , I'll test it again

**Comment by narayan954** [comment]
> … toggling and 3)adjusted the visibility of darkmode button on mobile screens

I'll be converting this pr to level 3 if it's okay? as you've solved multiple issues

**Comment by gitpugal** [comment]
yeah that'd be fine!.

File: src/lib/firebase.js (hunk 2 above)
Comment (gitpugal) [reply src/lib/firebase.js:15]: Yup, not needed as we won't be using google analytics.

File: src/pages/Profile/index.jsx (hunk 3 above)
Comment (gitpugal) [reply src/pages/Profile/index.jsx:71]: Yup I've added!

File: src/pages/Signup/index.jsx (hunk 4 above)
Comment (gitpugal) [reply src/pages/Signup/index.jsx:117]: yeah that's fine.

File: src/pages/Signup/index.jsx (hunk 5 above)
Comment (gitpugal) [reply src/pages/Signup/index.jsx:116]: Yup, Done!

File: src/pages/Signup/index.jsx (hunk 6 above)
Comment (gitpugal) [reply src/pages/Signup/index.jsx:204]: Done!.

File: src/components/ImgUpload/index.jsx (hunk 1 above)
Comment (gitpugal) [reply src/components/ImgUpload/index.jsx:97]: That's Done.

File: src/lib/firebase.js (hunk 2 above)
Comment (narayan954) [reply src/lib/firebase.js]: cool

**Comment by gitpugal** [comment]
@narayan954  Can you please now check the requested changes, I've resolved them as you've mentioned.I've resolved them as you've mentioned.

**Comment by narayan954** [comment]
> @narayan954 Can you please now check the requested changes, I've resolved them as you've mentioned.

sure, i'll check today

**Comment by gitpugal** [comment]
Good afternoon @narayan954 , Can I know the status of the PR 613 and if any changes needed from my side.

**Comment by narayan954** [comment]
> Good afternoon @narayan954 , Can I know the status of the PR 613 and if any changes needed from my side.

hi @gitpugal greetings
actually on signup using oauth, user is not created in the user collection... I'm looking into it

**Comment by narayan954** [comment]
> > Good afternoon @narayan954 , Can I know the status of the PR 613 and if any changes needed from my side.
> 
> hi @gitpugal greetings actually on signup using oauth, user is not created in the user collection... I'm looking into it

it's working,merging this

**Review by narayan954 — APPROVED** [review]
looks good to me!

**Comment by gitpugal** [comment]
Good to hear this and this is my first ever PR, I am really happy and thank you for assisting me humbly. thank you much @narayan954

**Pull request merged**